### PR TITLE
docs(effect): add recipe for useAtomEffect

### PR DESCRIPTION
**Pull request by dmaskasky**
## Summary
Adds a doc for useAtomEffect under the doc [Custom useAtom hooks](https://jotai.org/docs/recipes/custom-useatom-hooks).

## Check List

- [ ] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/recipes/useAtomEffect.mdx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: useAtomEffect
```
Comment (dai-shi) [inline docs/recipes/useAtomEffect.mdx:2]: Do you want to add a link to `extensions/effect.mdx`?
Or, I was thinking of adding this recipe there.

Comment (dmaskasky) [reply docs/recipes/useAtomEffect.mdx:2]: A link would be good.

**Review by dai-shi — COMMENTED** [review]
You commit two files.

**Comment by dmaskasky** [comment]
> You commit two files.

Fixed.

File: docs/recipes/use-atom-effect.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { useMemo } from 'react'
+import { useAtomValue } from 'jotai/react'
+import { atomEffect } from 'jotai-effect'
+
+type CleanupFn = () => void
+
+type EffectFn = (
+  get: Getter & { peek: Getter },
+  set: Setter & { recurse: Setter },
+) => CleanupFn | void
```
Comment (dai-shi) [inline docs/recipes/use-atom-effect.mdx:21]: Can this be:
```ts
type EffectFn = Parameters<typeof atomEffect>[0]
```

File: docs/recipes/use-atom-effect.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  (channelId: string) => {
+    return atom(new Channel(channelId))
+  },
+)
+const messagesAtom = atom<Message[]>([])
+
+function Messages({ channelId }: { channelId: string }) {
+  const [messages] = useAtom(messagesAtom)
+  useAtomEffect(
+    useCallback(
```
Comment (dai-shi) [inline docs/recipes/use-atom-effect.mdx:45]: Maybe we should add a small note about useCallback/useMemo that they aren't for semantic guarantee by React. So, it **may** run the effect twice **in the future**. (But, if it really happens, we could just rewrite the recipe using useRef based memo/callback. For example: https://github.com/alexreardon/use-memo-one )

Comment (dmaskasky) [reply docs/recipes/use-atom-effect.mdx:45]: Good point. I think the recipe should use the stable useCallback from use-memo-one and include this general note on useCallback.

**Review by dai-shi — APPROVED** [review]
LGTM

Now, it makes more sense to be a recipe instead of a library util.

**Pull request merged**